pysatCDAAC/instruments/methods/general.py

```diff
@@ -96,14 +105,75 @@ def download(date_array, tag, inst_id, supported_tags=None,
         try:
             # Uncompress files and remove tarball
             tar = tarfile.open(fname)
-            tar.extractall(path=os.path.join(data_path, yrdoystr))
+            # Note: sort_files only used for cosmic2 ivm
+            if sort_files:
+                inst_ids = ['e1', 'e2', 'e3', 'e4', 'e5', 'e6']
+                fnames = tar.getnames()
+                for id in inst_ids:
+                    search_str, final_path = get_instrument_data_info(
+                        'cosmic2_ivm', tag=tag, inst_id=id)
+                    matched_files = fnmatch.filter(fnames, search_str)
+                    for fname in matched_files:
+                        tar.extract(fname, path=final_path)
+            else:
+                if sub_path:
+                    final_path = os.path.join(data_path, yrdoystr)
+                else:
+                    final_path = data_path
+                tar.extractall(path=final_path)
```

Comment:
This seems out of place in a general function.  Would it not be better to just put this in the COSMIC2 IVM instrument?

Reply:
the alternative is to extract data to the wrong directories then clean up later